spec/lib/salus/processor_spec.rb

```diff
@@ -215,5 +227,64 @@ def remove_key(json_string, key = 'running_time')
         processor.export_report
       end
     end
+
+    context 'remote URI headers verbs' do
+      let(:expected_report) do
+        File.read('spec/fixtures/processor/remote_uri_headers_verbs/expected_report.json').strip
+      end
+
+      let(:remote_uri) { 'https://nerv.tk3/salus-report' }
+      let(:remote_uri1) { 'https://nerv.tk4/salus-report' }
+
+      it 'should send the report to the remote URI with correct headers and verb' do
+        stub_request(:put, remote_uri)
+          .with(headers: { 'Content-Type' => 'application/json',
+                           'X-API-Key' => '',
+                           'repo' => 'Random Repo' },
+                body: {})
+          .to_return(status: 202)
+
+        stub_request(:post, remote_uri1)
+          .with(headers: { 'Content-Type' => 'application/json',
+                           'X-API-Key' => '',
+                           'repo' => 'Random Repo' },
+                body: {})
+          .to_return(status: 202)
+
+        processor = Salus::Processor.new(
+          repo_path: 'spec/fixtures/processor/remote_uri_headers_verbs'
+        )
+        processor.scan_project
+        processor.export_report
+
+        assert_requested(
+          :put,
+          remote_uri,
+          headers:
+            {
+              'Content-Type' => 'application/json',
+              'X-API-Key' => '',
+              'repo' => 'Random Repo'
+            },
+          times: 1
+        ) do |req|
+          expect(req.body).to match_cyclonedx_report_json(expected_report)
+        end
+
+        assert_requested(
+          :post,
+          remote_uri1,
+          headers:
+            {
+              'Content-Type' => 'application/json',
+              'X-API-Key' => '',
+              'repo' => 'Random Repo'
+            },
+          times: 1
+        ) do |req|
+          expect(req.body).to match_cyclonedx_report_json(expected_report)
```

Comment:
Are there specific fields in the bom that require you to drop the bom from the cyclonedx match check here?

Reply:
Yep, the UUID field causes the base64 encoding to be different. I may be able to decode, remove that field and encode again 🤔